pkg/controller/jaeger/jaeger_controller_test.go

```diff
@@ -89,6 +95,71 @@ func TestDeletedInstance(t *testing.T) {
 	assert.Empty(t, persisted.Name) // this means that the object wasn't found
 }
 
+func TestCleanFinalizer(t *testing.T) {
+	jaeger := v1.NewJaeger("TestDeletedInstance")
+	dep := appsv1.Deployment{
+		Spec: appsv1.DeploymentSpec{
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{},
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{
+						corev1.Container{},
+					},
+				},
+			},
+		},
+	}
+	dep.Name = "mydep"
+	dep.Annotations = map[string]string{inject.Annotation: jaeger.Name}
+	s := scheme.Scheme
+	s.AddKnownTypes(v1.SchemeGroupVersion, jaeger)
+
+	jaeger.DeletionTimestamp = &metav1.Time{Time: time.Now()}
+	jaeger.SetFinalizers([]string{finalizer})
+	cl := fake.NewFakeClient(jaeger, inject.Sidecar(jaeger, &dep))
+
+	r := &ReconcileJaeger{client: cl, scheme: nil}
+
+	req := reconcile.Request{
+		NamespacedName: types.NamespacedName{
+			Name:      jaeger.Name,
+			Namespace: jaeger.Namespace,
+		},
+	}
+
+	// execute finalizer
+	_, err := r.Reconcile(req)
+
+	// verify
+	assert.NoError(t, err)
+	persisted := &appsv1.Deployment{}
+	err = cl.Get(context.Background(), types.NamespacedName{
+		Namespace: dep.Namespace,
+		Name:      dep.Name,
+	}, persisted)
+	assert.Equal(t, len(persisted.Spec.Template.Spec.Containers), 1)
```

Comment:
Can you add an assertion on the annotation as well? The cleanup procedure should remove the `inject.Annotation` from the deployment as well.